### PR TITLE
Add agent version filter option

### PR DESCRIPTION
**Pull request by GGP1**
|Parent issue|Source issue|Related issue | Related PR|
|---|---|---|---|
| https://github.com/wazuh/wazuh/issues/18138 | https://github.com/wazuh/wazuh/issues/11011 | https://github.com/wazuh/wazuh/issues/18139 | https://github.com/wazuh/wazuh/pull/18212 |

## Description

Added the `agents -> allow_higher_versions -> allow` option to the API configuration documentation page.

> I used the default value and range indicated in https://github.com/wazuh/wazuh/issues/18138

### Screenshots

schema

config

### CODE REVIEW

File: source/user-manual/api/configuration.rst [hunk 1]
```diff
@@ -324,6 +327,20 @@ limits
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, no              | yes          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section inside the :ref:`global tag <reference_ossec_global>`.                                                                                 |
```
Comment (Selutario) [inline source/user-manual/api/configuration.rst:341]:
```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with value ``yes`` inside the :ref:`global tag <reference_ossec_global>`.                                       |
```

**Review by Selutario — CHANGES_REQUESTED** [review]
The table is malformed:
image
image

File: source/user-manual/api/configuration.rst [hunk 2]
```diff
@@ -324,6 +327,20 @@ limits
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, true, no, false | true          | Allow uploading configurations accepting higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                   |
```
Comment (s-ocando) [inline source/user-manual/api/configuration.rst:341]:
```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                 |
```

File: source/user-manual/api/configuration.rst [hunk 3]
```diff
@@ -324,6 +327,20 @@ limits
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<agents><allow_higher_versions>`` section with the ``yes`` value inside the :ref:`global tag <reference_ossec_global>`.                                 |
```
Comment (Selutario) [inline source/user-manual/api/configuration.rst:341]: After the last changes made [here](https://github.com/wazuh/wazuh/pull/18342), this part needs to be updated:

```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations that accept higher agent versions through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the ``<allow_higher_versions>`` section with the ``yes`` value inside the :ref:`auth <reference_ossec_auth>` or :ref:`remote <reference_ossec_remote>` tags.  |
```

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by s-ocando — APPROVED** [review]

**Review by Selutario — APPROVED** [review]

**Pull request merged**